packages/core/test/acceptance/sequence/sequence.acceptance.ts

```diff
@@ -47,7 +47,7 @@ describe('Sequence - ', () => {
       }
     }
     // bind user defined sequence
-    app.bind('sequence').toClass(MySequence);
+    app.sequence(MySequence);
```

Comment:
Interestingly, getting this:

```
[ts]
Argument of type 'typeof MySequence' is not assignable to parameter of type 'Constructor<Sequence>'.
  Type 'MySequence' is not assignable to type 'Sequence'.
    Property 'findRoute' is protected but type 'MySequence' is not a class derived from 'Sequence'.
```

This makes me think maybe we should refactor `Sequence` into either an Abstract class or Interface and move our current `Sequence` impl into `DefaultSequence`. Thoughts? @raymondfeng @ritch @bajtos @rashmihunt